### ISSUE TITLE
Implement Kálmán filter - transformer (with update functionality)

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1011]
The vanilla Kálmán filter should be implemented or interfaced as a (potentially multivariate) single-series transformer.

**Comment by lielleravid** [comment]
Hi! We would like to work on this issue :)

**Comment by aiwalter** [comment]
cool, you might also look into the draft PR #1222

**Comment by lielleravid** [comment]
Also, how would you like us to take care of the case where there is missing measurements in the data? Is this something we should take into account?

**Comment by fkiraly** [comment]
@lielleravid, I think optimally the user can choose which of these matrices are user input and which are estimated, with a default of estimation.
I would also advise to start with investigating potential packages that could be interfaced here - Kalman filters are not uncommon, although I do not have a full view of what is out there.

**Comment by serhan-gul** [comment]
@lielleravid You might want to check the implementation in [FilterPy](https://filterpy.readthedocs.io/en/latest/). It is typical that the noice matrices Q and R are specified by the user since they depend on the considered system.

**Comment by fkiraly** [comment]
Yes, @lielleravid and @NoaBenAmi are both working on this currently, and looking at `filterpy` and `pykalman` to interface. One of the two (I forgot which) has more capability to estimate the various matrices.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues this week — I think this one has shipped and can be closed.

The Kalman filter transformer landed (the issue body itself notes this), and three backends are now wired up under `sktime/transformations/series/kalman_filter/`:

- [`_kalman_filter.py:130`](https://github.com/sktime/sktime/blob/main/sktime/transformations/series/kalman_filter/_kalman_filter.py#L130) — `KalmanFilterTransformerPK` (pykalman backend).
- [`_kalman_filter.py:630`](https://github.com/sktime/sktime/blob/main/sktime/transformations/series/kalman_filter/_kalman_filter.py#L630) — `KalmanFilterTransformerFP` (FilterPy backend).
- [`_simdkalman.py:12`](https://github.com/sktime/sktime/blob/main/sktime/transformations/series/kalman_filter/_simdkalman.py#L12) — `KalmanFilterTransformerSIMD` (simdkalman backend).

Tests live in `sktime/transformations/series/tests/test_kalman_filter.py`. The remaining nuance the issue mentioned — explicit `update` functionality — looks like it would be better tracked as a fresh, scoped issue against the current implementations rather than left on this 2021 ticket.

Would you mind closing this out? If a specific `update`-related gap is blocking you, happy to look further.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against the current `main` branch at commit `bd8fe08`.)
